### PR TITLE
Use elemHashes in lookup

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: bloomfilter/src/Data/BloomFilter/Hash.hs [hunk 1]
```diff
@@ -173,6 +177,38 @@ instance Hashes CheapHashes where
     evalHashes = evalCheapHashes
     {-# INLINE evalHashes #-}
 
+instance Prim (CheapHashes a) where
+    sizeOfType# _ = 128#
+    alignmentOfType# _ = 64#
```
Comment (jorisdral) [inline bloomfilter/src/Data/BloomFilter/Hash.hs:182]: These are in number of bytes

```suggestion
    sizeOfType# _ = 16#
    alignmentOfType# _ = 8#
```

I didn't immediately spot this until I ran `primLaws` tests from `quickcheck-classes`. Feel free to include this in the tests if you think it's useful:

```haskell

-------------------------------------------------------------------------------
-- Prim
-------------------------------------------------------------------------------

instance Arbitrary (CheapHashes a) where
  arbitrary = CheapHashes <$> arbitrary <*> arbitrary
  shrink (CheapHashes x y) = uncurry CheapHashes <$> shrink (x, y)

deriving instance Eq (CheapHashes a)

prop_primLaws :: Property
prop_primLaws = ioProperty $
    lawsCheck $ primLaws (Proxy @(CheapHashes Void))

testLawsCheck :: Laws -> TestTree
testLawsCheck (Laws className properties) =
    testGroup className [
        testProperty name prop
      | (name, prop) <- properties
      ]
```

**Review by jorisdral — CHANGES_REQUESTED** [review]
Net time for bloom filter queries in the lookup macro-benchmark went from 3.55s down to 1.93s. That's a pretty good improvement!

**Review by jorisdral — APPROVED** [review]
Let's squash and merge

**Comment by dcoutts** [comment]
I tried using this in the macrobenchmark for bloom filters only (not the whole lookup) and get odd results with the allocations. The time is good though, and faster than the one-by-one. That mystery doesn't have to be solved for this PR though.

**Pull request merged**